server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPRulesRoutesTest.java

```diff
@@ -0,0 +1,410 @@
+                .containsOnly("3");
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenIdIsNotSpecified() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITHOUT_ID)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("id field is mandatory"));
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenExpressionIsNotSpecified() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITHOUT_EXPRESSION)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("expression field is mandatory"));
+        }
+
+        @Test
+        void putShouldReturnNotFoundWhenDomainNotInList() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITH_POSITIVE_TARGETS)
+                .add(JSON_WITH_NEGATIVE_TARGETS)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put("strange.com")
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(404))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("The domain list does not contain: strange.com"));
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenDomainIsNotValid() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITH_POSITIVE_TARGETS)
+                .add(JSON_WITH_NEGATIVE_TARGETS)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put("dr@strange.com")
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("Invalid request for domain dr@strange.com"));
+        }
+    }
+
+    @Nested
+    class DefineClear {
+
+        @Test
+        void deleteShouldRemoveTheConfigurations() {
+            dlpStore.store(SENDER_DOMAIN, CONFIGURATION_ITEMS);
```

Comment:
Mhh

You are using a lower layer in order to test the webadmin layer. This means you corelate the webadmin test suite with it's implementation (here the Configuration store) and make changing the implementation harder.

We try hard to use the same layer in our test, and avoid cross layer testing...

Can't you use a PUT, and LIST to replace direct calls to the dlpStore everywhere in this test class?